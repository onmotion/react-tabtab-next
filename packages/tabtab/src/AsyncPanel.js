"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var Panel_1 = __importDefault(require("./Panel"));
var AsyncPanelComponent = (function (_super) {
    __extends(AsyncPanelComponent, _super);
    function AsyncPanelComponent(props) {
        var _this = _super.call(this, props) || this;
        _this.loadPanel = _this.loadPanel.bind(_this);
        _this.cacheData = undefined;
        _this.state = {
            isLoading: false,
            data: undefined
        };
        return _this;
    }
    AsyncPanelComponent.prototype.componentDidMount = function () {
        if (this.props.active)
            this.loadPanel();
    };
    AsyncPanelComponent.prototype.componentWillReceiveProps = function (nextProps) {
        if (nextProps.active)
            this.loadPanel();
    };
    AsyncPanelComponent.prototype.loadPanel = function () {
        var _this = this;
        var _a = this.props, loadContent = _a.loadContent, cache = _a.cache;
        if (cache && this.cacheData) {
            this.setState({
                isLoading: false,
                data: this.cacheData
            });
            return;
        }
        var callback = function (err, data) {
            if (err) {
                console.log('React-Tabtab async panel error:', err);
            }
            if (cache) {
                _this.cacheData = data;
            }
            _this.setState({
                isLoading: false,
                data: data
            });
        };
        var promise = loadContent(callback);
        if (promise) {
            promise.then(function (data) { return callback(null, data); }, function (err) { return callback(err); });
        }
        if (!this.state.isLoading) {
            this.setState({ isLoading: true });
        }
    };
    AsyncPanelComponent.prototype.render = function () {
        var _a = this.props, render = _a.render, renderLoading = _a.renderLoading, CustomPanelStyle = _a.CustomPanelStyle, active = _a.active, index = _a.index;
        var _b = this.state, isLoading = _b.isLoading, data = _b.data;
        var content;
        if (isLoading) {
            content = renderLoading();
        }
        else {
            content = render(data);
        }
        return (React.createElement(Panel_1.default, __assign({}, { CustomPanelStyle: CustomPanelStyle, active: active, index: index }), content));
    };
    AsyncPanelComponent.defaultProps = {
        cache: true
    };
    return AsyncPanelComponent;
}(React.PureComponent));
exports.default = AsyncPanelComponent;
