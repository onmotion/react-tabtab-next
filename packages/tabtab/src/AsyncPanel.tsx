import * as React from 'react';
import Panel, { PanelProps } from './Panel';

type Props = {
    loadContent: (cb: (err: any, data?: any) => void) => any;
    render: (data: any) => React.ReactNode;
    renderLoading: () => React.ReactNode;
    CustomPanelStyle?: React.FC<Partial<PanelProps>>;
    active?: boolean;
    index?: number;
    cache?: boolean;
};

type State = {
    isLoading: boolean;
    data: any;
};

export default class AsyncPanelComponent extends React.PureComponent<Props, State> {
    static defaultProps = {
        cache: true,
    };

    cacheData: any;

    constructor(props: Props) {
        super(props);
        this.loadPanel = this.loadPanel.bind(this);
        this.cacheData = undefined;
        this.state = {
            isLoading: false,
            data: undefined,
        };
    }

    componentDidMount() {
        if (this.props.active) this.loadPanel();
    }
    componentDidUpdate(prevProps: Props) {
        this.props.active && !prevProps.active && this.loadPanel();
    }

    loadPanel() {
        console.log('loadPanel');

        const { loadContent, cache } = this.props;
        if (cache && this.cacheData) {
            this.setState({
                isLoading: false,
                data: this.cacheData,
            });
            return;
        }
        const callback = (err: any, data?: any) => {
            if (err) {
                console.log('React-Tabtab async panel error:', err);
            }
            if (cache) {
                this.cacheData = data;
            }
            this.setState({
                isLoading: false,
                data,
            });
        };
        const promise = loadContent(callback);
        if (promise) {
            promise.then(
                (data: any) => callback(null, data),
                (err: any) => callback(err)
            );
        }
        if (!this.state.isLoading) {
            this.setState({ isLoading: true });
        }
    }

    render() {
        const { render, renderLoading, CustomPanelStyle, active, index } = this.props;
        const { isLoading, data } = this.state;
        let content;
        if (isLoading) {
            content = renderLoading();
        } else {
            content = render(data);
        }
        return <Panel {...{ CustomPanelStyle, active, index }}>{content}</Panel>;
    }
}
