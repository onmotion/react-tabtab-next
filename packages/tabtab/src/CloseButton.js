"use strict";
var __makeTemplateObject = (this && this.__makeTemplateObject) || function (cooked, raw) {
    if (Object.defineProperty) { Object.defineProperty(cooked, "raw", { value: raw }); } else { cooked.raw = raw; }
    return cooked;
};
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var IconSvg_1 = require("./IconSvg");
var styled_components_1 = __importDefault(require("styled-components"));
var CloseWrapper = styled_components_1.default.button(templateObject_1 || (templateObject_1 = __makeTemplateObject(["\n  display: inline-block;\n  color: #777;\n  margin-left: 5px;\n  padding: 0;\n  vertical-align: middle;\n  border: 0;\n  padding: 2px;\n  outline: 0;\n  &:hover {\n    color: black;\n    background-color: #eee;\n    cursor: pointer;\n    border-radius: 50%;\n  }\n  > svg {\n    vertical-align: middle;\n  }\n"], ["\n  display: inline-block;\n  color: #777;\n  margin-left: 5px;\n  padding: 0;\n  vertical-align: middle;\n  border: 0;\n  padding: 2px;\n  outline: 0;\n  &:hover {\n    color: black;\n    background-color: #eee;\n    cursor: pointer;\n    border-radius: 50%;\n  }\n  > svg {\n    vertical-align: middle;\n  }\n"])));
var CloseButton = (function (_super) {
    __extends(CloseButton, _super);
    function CloseButton() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    CloseButton.prototype.render = function () {
        return (React.createElement(CloseWrapper, { onClick: this.props.handleDelete },
            React.createElement(IconSvg_1.CloseIcon, null)));
    };
    return CloseButton;
}(React.PureComponent));
exports.default = CloseButton;
var templateObject_1;
