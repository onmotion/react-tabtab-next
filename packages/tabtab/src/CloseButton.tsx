// @flow
import * as React from 'react';
import { CloseIcon } from './IconSvg';
import styled from 'styled-components';

const CloseWrapper = styled.button`
    display: inline-block;
    color: #777;
    line-height: 0;
    margin-left: 5px;
    padding: 0;
    vertical-align: middle;
    background-color: transparent;
    border: 0;
    padding: 2px;
    outline: 0;
    &:hover {
        color: black;
        background-color: #eee;
        cursor: pointer;
        border-radius: 50%;
    }
    > svg {
        vertical-align: middle;
    }
`;

type Props = {
    handleTabClose: (event: any) => void;
};

export default class CloseButton extends React.PureComponent<Props> {
    render() {
        return (
            <CloseWrapper onClick={this.props.handleTabClose}>
                <CloseIcon />
            </CloseWrapper>
        );
    }
}
