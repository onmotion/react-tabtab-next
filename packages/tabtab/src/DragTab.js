"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        if (typeof b !== "function" && b !== null)
            throw new TypeError("Class extends value " + String(b) + " is not a constructor or null");
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __rest = (this && this.__rest) || function (s, e) {
    var t = {};
    for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p) && e.indexOf(p) < 0)
        t[p] = s[p];
    if (s != null && typeof Object.getOwnPropertySymbols === "function")
        for (var i = 0, p = Object.getOwnPropertySymbols(s); i < p.length; i++) {
            if (e.indexOf(p[i]) < 0 && Object.prototype.propertyIsEnumerable.call(s, p[i]))
                t[p[i]] = s[p[i]];
        }
    return t;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
var React = __importStar(require("react"));
var react_sortable_hoc_1 = require("react-sortable-hoc");
var Tab_1 = __importDefault(require("./Tab"));
var DragTabElement = (0, react_sortable_hoc_1.SortableElement)(function (_a) {
    var children = _a.children, props = __rest(_a, ["children"]);
    return (React.createElement(Tab_1.default, __assign({}, props, { index: props.tabIndex }), children));
});
var DragTab = (function (_super) {
    __extends(DragTab, _super);
    function DragTab() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    DragTab.prototype.render = function () {
        var _this = this;
        var _a = this.props, children = _a.children, props = __rest(_a, ["children"]);
        return (React.createElement(DragTabElement, __assign({ index: 0, ref: function (node) { return _this.__DRAG_TAB_INTERNAL_NODE = node; } }, props), children));
    };
    return DragTab;
}(React.PureComponent));
exports.default = DragTab;
