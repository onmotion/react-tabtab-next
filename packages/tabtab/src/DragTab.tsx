// @flow
import React, { FC, forwardRef, memo } from 'react';
import { useSortable } from '@dnd-kit/sortable';
import { CSS } from '@dnd-kit/utilities';

interface Props {
    id: string;
    activeIndex?: number;
    index?: number;
    children: React.ReactNode;
}
const DragTab: FC<Props> = memo(
    forwardRef<HTMLElement, Props>(({ children, id, index, activeIndex, ...rest }, ref) => {
        const { attributes, listeners, setNodeRef, transform, transition } = useSortable({ id: id });

        const style = {
            transform: CSS.Transform.toString(transform),
            transition,
            // cursor: 'default',
        };

        return (
            <div ref={setNodeRef} style={style} {...attributes} {...listeners}>
                {React.cloneElement(children as React.ReactElement, {
                    ...rest,
                    key: id,
                    active: index === activeIndex,
                    index,
                    tabIndex: id,
                    ref: ref,
                })}
            </div>
        );
    })
);

export default DragTab;
