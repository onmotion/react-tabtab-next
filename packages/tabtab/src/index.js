"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.styled = exports.ExtraButton = exports.AsyncPanel = exports.Panel = exports.PanelList = exports.DragTab = exports.DragTabList = exports.Tab = exports.TabList = exports.Tabs = void 0;
var Tabs_1 = __importDefault(require("./Tabs"));
exports.Tabs = Tabs_1.default;
var TabList_1 = __importStar(require("./TabList"));
exports.TabList = TabList_1.default;
var Tab_1 = __importStar(require("./Tab"));
exports.Tab = Tab_1.default;
var DragTabList_1 = __importDefault(require("./DragTabList"));
exports.DragTabList = DragTabList_1.default;
var DragTab_1 = __importDefault(require("./DragTab"));
exports.DragTab = DragTab_1.default;
var PanelList_1 = __importDefault(require("./PanelList"));
exports.PanelList = PanelList_1.default;
var Panel_1 = __importStar(require("./Panel"));
exports.Panel = Panel_1.default;
var AsyncPanel_1 = __importDefault(require("./AsyncPanel"));
exports.AsyncPanel = AsyncPanel_1.default;
var ExtraButton_1 = __importDefault(require("./ExtraButton"));
exports.ExtraButton = ExtraButton_1.default;
exports.styled = { TabListStyle: TabList_1.TabListStyle, ActionButtonStyle: TabList_1.ActionButtonStyle, TabStyle: Tab_1.TabStyle, PanelStyle: Panel_1.PanelStyle };
