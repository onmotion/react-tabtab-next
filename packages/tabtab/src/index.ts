// @flow
import Tabs from "./Tabs"
import TabList, { TabListStyle, ActionButtonStyle } from "./TabList"
import Tab, { TabStyle } from "./Tab"
import DragTabList from "./DragTabList"
import DragTab from "./DragTab"
import PanelList from "./PanelList"
import Panel, { PanelStyle } from "./Panel"
import AsyncPanel from "./AsyncPanel"
import ExtraButton from "./ExtraButton"

export {
  Tabs,
  TabList,
  Tab,
  DragTabList,
  DragTab,
  PanelList,
  Panel,
  AsyncPanel,
  ExtraButton
}

export const styled = { TabListStyle, ActionButtonStyle, TabStyle, PanelStyle }
