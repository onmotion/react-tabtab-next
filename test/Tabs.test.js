import React from 'react';
import {Tabs, DragTabList, DragTab, TabList, Tab, PanelList, Panel, ExtraButton} from '../src';
import {shallow, mount} from 'enzyme';

const normalComponent = (props = {}) => (
  <Tabs {...props}>
    <TabList>
      <Tab>Tab1</Tab>
      <Tab>Tab2</Tab>
    </TabList>
    <PanelList>
      <Panel>Content1</Panel>
      <Panel>Content2</Panel>
    </PanelList>
  </Tabs>
);

const dragComponent = (props = {}) => (
  <Tabs {...props}>
    <DragTabList>
      <DragTab>DragTab1</DragTab>
      <DragTab>DragTab2</DragTab>
    </DragTabList>
    <PanelList>
      <Panel>Content1</Panel>
      <Panel>Content2</Panel>
    </PanelList>
  </Tabs>
);

describe('render Tabs', () => {
  const testRender = (tabsComponent) => {
    const component = mount(tabsComponent);
    expect(component.html()).toMatchSnapshot();
  }
  it('normal tabs', () => {
    testRender(normalComponent())
  });
  it('normal drag tabs', () => {
    testRender(dragComponent());
  });
  it('normal tab with ExtraButton', () => {
    testRender(normalComponent({
      ExtraButton: 
        <ExtraButton>
          +
        </ExtraButton>
    }))
  })
  it('normal drag with ExtraButton', () => {
    testRender(dragComponent({
      ExtraButton:
        <ExtraButton>
          +
        </ExtraButton>
    }))
  })
});

describe('props', () => {
  it('defaultIndex', () => {
    const defaultIndex = 1;
    const component = shallow(normalComponent({defaultIndex}));
    expect(component.state().activeIndex).toEqual(defaultIndex);
  })

  it('click tab, onTabChange callback', () => {
    const mockTabChange = jest.fn();
    const component = mount(normalComponent({onTabChange: mockTabChange}));
    component.find('Tab').at(1).simulate('click');
    expect(component.state().activeIndex).toEqual(1);
    expect(mockTabChange).toBeCalled();
    expect(mockTabChange.mock.calls[0][0]).toEqual(1);
  })

  describe('activeIndex', () => {
    it('show active index', () => {
      const mountComponent = mount(normalComponent({activeIndex: 1}));
      expect(mountComponent.state().activeIndex).toEqual(1);
    });

    it('do nothing when click tab', () => {
      const mockTabChange = jest.fn();
      const mountComponent = mount(normalComponent({onTabChange: mockTabChange, activeIndex: 1}));
      mountComponent.find('Tab').at(0).simulate('click');
      expect(mountComponent.state().activeIndex).toEqual(1);
    });

    it('update active tab when pass new activeKey', () => {
      const mockTabChange = jest.fn();
      const mountComponent = mount(normalComponent({onTabChange: mockTabChange, activeIndex: 1}));
      mountComponent.setProps({activeIndex: 0})
      expect(mountComponent.state().activeIndex).toEqual(0);
    })
  })
})
