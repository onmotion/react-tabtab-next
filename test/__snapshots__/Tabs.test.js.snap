// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Tabs normal drag tabs 1`] = `"<div><div><div class=\\"sc-bdVaJa fzhJWH\\"><div><div class=\\"sc-fjdhpX gAbfpE\\"><svg viewBox=\\"0 0 40 40\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" style=\\"vertical-align: middle;\\"><g><path d=\\"m25.7 12.3l-7.7 7.7 7.7 7.7-2.3 2.3-10-10 10-10z\\"></path></g></svg></div><div class=\\"sc-fjdhpX gYthMW\\"><svg viewBox=\\"0 0 40 40\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" style=\\"vertical-align: middle;\\"><g><path d=\\"m16.6 10l10 10-10 10-2.3-2.3 7.7-7.7-7.7-7.7z\\"></path></g></svg></div></div><div class=\\"sc-bwzfXH fvdpZL\\"><ul role=\\"tablist\\" class=\\"sc-htpNat ezotGS\\"><li role=\\"tab\\" id=\\"react-tabtab-tab-0\\" aria-controls=\\"react-tabtab-panel-0\\" aria-selected=\\"true\\" class=\\"sc-EHOje cszVDW\\"><span class=\\"sc-bZQynM gIZEjL\\">DragTab1</span></li><li role=\\"tab\\" id=\\"react-tabtab-tab-1\\" aria-controls=\\"react-tabtab-panel-1\\" aria-selected=\\"false\\" class=\\"sc-EHOje cszVDW\\"><span class=\\"sc-bZQynM gIZEjL\\">DragTab2</span></li></ul></div></div></div><div><div role=\\"tabpanel\\" id=\\"react-tabtab-panel-0\\" aria-labelledby=\\"react-tabtab-0\\" aria-hidden=\\"false\\" class=\\"sc-gzVnrw jnZyxb\\">Content1</div><div role=\\"tabpanel\\" id=\\"react-tabtab-panel-1\\" aria-labelledby=\\"react-tabtab-1\\" aria-hidden=\\"false\\" class=\\"sc-gzVnrw csBXJJ\\"></div></div></div>"`;

exports[`render Tabs normal drag with ExtraButton 1`] = `"<div><div><button class=\\"sc-htoDjs dmTOMv\\">+</button><div class=\\"sc-bdVaJa fzhJWH\\"><div><div class=\\"sc-dxgOiQ diAQHo\\"><svg viewBox=\\"0 0 40 40\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" style=\\"vertical-align: middle;\\"><g><path d=\\"m25.7 12.3l-7.7 7.7 7.7 7.7-2.3 2.3-10-10 10-10z\\"></path></g></svg></div><div class=\\"sc-dxgOiQ fusejr\\"><svg viewBox=\\"0 0 40 40\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" style=\\"vertical-align: middle;\\"><g><path d=\\"m16.6 10l10 10-10 10-2.3-2.3 7.7-7.7-7.7-7.7z\\"></path></g></svg></div></div><div class=\\"sc-bwzfXH fvdpZL\\"><ul role=\\"tablist\\" class=\\"sc-htpNat ezotGS\\"><li role=\\"tab\\" id=\\"react-tabtab-tab-0\\" aria-controls=\\"react-tabtab-panel-0\\" aria-selected=\\"true\\" class=\\"sc-EHOje cszVDW\\"><span class=\\"sc-bZQynM gIZEjL\\">DragTab1</span></li><li role=\\"tab\\" id=\\"react-tabtab-tab-1\\" aria-controls=\\"react-tabtab-panel-1\\" aria-selected=\\"false\\" class=\\"sc-EHOje cszVDW\\"><span class=\\"sc-bZQynM gIZEjL\\">DragTab2</span></li></ul></div></div></div><div><div role=\\"tabpanel\\" id=\\"react-tabtab-panel-0\\" aria-labelledby=\\"react-tabtab-0\\" aria-hidden=\\"false\\" class=\\"sc-gzVnrw jnZyxb\\">Content1</div><div role=\\"tabpanel\\" id=\\"react-tabtab-panel-1\\" aria-labelledby=\\"react-tabtab-1\\" aria-hidden=\\"false\\" class=\\"sc-gzVnrw csBXJJ\\"></div></div></div>"`;

exports[`render Tabs normal tab with ExtraButton 1`] = `"<div><div><button class=\\"sc-htoDjs dmTOMv\\">+</button><div class=\\"sc-bdVaJa fzhJWH\\"><div><div class=\\"sc-chPdSV fOysLs\\"><svg viewBox=\\"0 0 40 40\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" style=\\"vertical-align: middle;\\"><g><path d=\\"m25.7 12.3l-7.7 7.7 7.7 7.7-2.3 2.3-10-10 10-10z\\"></path></g></svg></div><div class=\\"sc-chPdSV exkSfS\\"><svg viewBox=\\"0 0 40 40\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" style=\\"vertical-align: middle;\\"><g><path d=\\"m16.6 10l10 10-10 10-2.3-2.3 7.7-7.7-7.7-7.7z\\"></path></g></svg></div></div><div class=\\"sc-bwzfXH fvdpZL\\"><ul role=\\"tablist\\" class=\\"sc-htpNat ezotGS\\"><li role=\\"tab\\" id=\\"react-tabtab-tab-0\\" aria-controls=\\"react-tabtab-panel-0\\" aria-selected=\\"true\\" class=\\"sc-EHOje cszVDW\\"><span class=\\"sc-bZQynM gIZEjL\\">Tab1</span></li><li role=\\"tab\\" id=\\"react-tabtab-tab-1\\" aria-controls=\\"react-tabtab-panel-1\\" aria-selected=\\"false\\" class=\\"sc-EHOje cszVDW\\"><span class=\\"sc-bZQynM gIZEjL\\">Tab2</span></li></ul></div></div></div><div><div role=\\"tabpanel\\" id=\\"react-tabtab-panel-0\\" aria-labelledby=\\"react-tabtab-0\\" aria-hidden=\\"false\\" class=\\"sc-gzVnrw jnZyxb\\">Content1</div><div role=\\"tabpanel\\" id=\\"react-tabtab-panel-1\\" aria-labelledby=\\"react-tabtab-1\\" aria-hidden=\\"false\\" class=\\"sc-gzVnrw csBXJJ\\"></div></div></div>"`;

exports[`render Tabs normal tabs 1`] = `"<div><div><div class=\\"sc-bdVaJa fzhJWH\\"><div><div class=\\"sc-gZMcBi kGTUtH\\"><svg viewBox=\\"0 0 40 40\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" style=\\"vertical-align: middle;\\"><g><path d=\\"m25.7 12.3l-7.7 7.7 7.7 7.7-2.3 2.3-10-10 10-10z\\"></path></g></svg></div><div class=\\"sc-gZMcBi fPUvwG\\"><svg viewBox=\\"0 0 40 40\\" fill=\\"currentColor\\" height=\\"1em\\" width=\\"1em\\" style=\\"vertical-align: middle;\\"><g><path d=\\"m16.6 10l10 10-10 10-2.3-2.3 7.7-7.7-7.7-7.7z\\"></path></g></svg></div></div><div class=\\"sc-bwzfXH fvdpZL\\"><ul role=\\"tablist\\" class=\\"sc-htpNat ezotGS\\"><li role=\\"tab\\" id=\\"react-tabtab-tab-0\\" aria-controls=\\"react-tabtab-panel-0\\" aria-selected=\\"true\\" class=\\"sc-EHOje cszVDW\\"><span class=\\"sc-bZQynM gIZEjL\\">Tab1</span></li><li role=\\"tab\\" id=\\"react-tabtab-tab-1\\" aria-controls=\\"react-tabtab-panel-1\\" aria-selected=\\"false\\" class=\\"sc-EHOje cszVDW\\"><span class=\\"sc-bZQynM gIZEjL\\">Tab2</span></li></ul></div></div></div><div><div role=\\"tabpanel\\" id=\\"react-tabtab-panel-0\\" aria-labelledby=\\"react-tabtab-0\\" aria-hidden=\\"false\\" class=\\"sc-gzVnrw jnZyxb\\">Content1</div><div role=\\"tabpanel\\" id=\\"react-tabtab-panel-1\\" aria-labelledby=\\"react-tabtab-1\\" aria-hidden=\\"false\\" class=\\"sc-gzVnrw csBXJJ\\"></div></div></div>"`;
